backend/src/main/java/com/redhat/cloud/notifications/auth/principal/rhid/RhServiceAccountIdentity.java

```diff
@@ -39,6 +39,11 @@ public String getName() {
         return getServiceAccount().getClientId();
     }
 
+    @Override
+    public String getUserId() {
+        return getServiceAccount().getClientId();
+    }
+
```

Comment:
[The service accounts will also have the `user_id` field](https://github.com/RedHatInsights/identity-schemas/blob/68c0a9f230f30cf93ce1ba80019fb0060e490ec6/3scale/identities/service-account.json#L74):

```json
{
    "identity": {
        "service_account": {
            "client_id": "b69eaf9e-e6a6-4f9e-805e-02987daddfbd",
            "username": "service-account-b69eaf9e-e6a6-4f9e-805e-02987daddfbd",
            "scope": "api.notifications",
            "user_id": "60ce65dc-4b5a-4812-8b65-b48178d92b12"
    }
}
```

So I think we need to do the same thing you did for the `RhUserIdentity` class.